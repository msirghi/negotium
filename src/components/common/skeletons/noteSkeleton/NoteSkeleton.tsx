import { Grid, Skeleton } from '@mui/material';
import { useNoteSkeletonStyles } from './styles';
import { ContentBox } from '../../boxes/content/ContentBox';
import { NoteSkeletonRow } from './row/NoteSkeletonRow';
import { useIsMobile } from '../../../../common/hooks/common/useIsMobile';

export const NoteSkeleton = () => {
  const classes = useNoteSkeletonStyles();
  const isMobile = useIsMobile();

  return (
    <ContentBox skipWidthChange>
      <div className={classes.inputContainer}>
        <Skeleton
          width={isMobile ? '100%' : '33%'}
          className={classes.inputSkeleton}
          height={50}
        />
      </div>
      <Grid container>
        <NoteSkeletonRow />
        <NoteSkeletonRow />
        <NoteSkeletonRow />
      </Grid>
    </ContentBox>
  );
};
