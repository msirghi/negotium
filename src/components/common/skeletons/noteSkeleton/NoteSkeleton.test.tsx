import renderer from 'react-test-renderer';
import { NoteSkeleton } from './NoteSkeleton';
import TestUtils from '../../../../common/tests/TestUtils';
import { mount } from 'enzyme';
import { Skeleton } from '@mui/material';

describe('NoteSkeleton', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<NoteSkeleton />);
    expect(tree).toMatchSnapshot();
  });

  it('should render skeleton input skeleton with with 100% on mobile', () => {
    window.matchMedia = TestUtils.createMatchMedia(500) as any;
    const wrapper = mount(<NoteSkeleton />);
    expect(wrapper.find(Skeleton).at(0).props().width).toEqual('100%');
  });

  it('should render skeleton input skeleton with with 33% on desktop', () => {
    window.matchMedia = TestUtils.createMatchMedia(1024) as any;
    const wrapper = mount(<NoteSkeleton />);
    expect(wrapper.find(Skeleton).at(0).props().width).toEqual('33%');
  });
});
