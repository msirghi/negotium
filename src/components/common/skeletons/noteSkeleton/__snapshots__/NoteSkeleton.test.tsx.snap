// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteSkeleton should match snapshot 1`] = `
<div
  className="makeStyles-root-5 makeStyles-root-6 MuiBox-root css-0"
>
  <div
    className="makeStyles-inputContainer-4"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-inputSkeleton-3 css-1n7gpw1-MuiSkeleton-root"
      style={
        Object {
          "height": 50,
          "width": "33%",
        }
      }
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-157tyuy-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse makeStyles-noteSkeleton-2 css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": 240,
            "width": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;
